packages/snfoundry/scripts-ts/verify-contracts.ts

```diff
@@ -0,0 +1,46 @@
+import path from 'path';
+import { execSync } from 'child_process';
+import yargs from 'yargs';
+
+// Import deployedContracts
+const deployedContractsPath = path.resolve(__dirname, '../../nextjs/contracts/deployedContracts');
+const deployedContracts = require(deployedContractsPath).default;
+
+// Parse command line arguments
+const argv = yargs(process.argv.slice(2))
+  .option('network', {
+    type: 'string',
+    description: 'Specify the network mainnet or sepolia',
+    demandOption: true,
+  })
+  .parseSync();
+
+const network = argv.network;
+
+if (!deployedContracts[network]) {
+  console.error(`No deployed contracts found for network: ${network}`);
+  process.exit(1);
+}
+
+// Change to the contracts directory
+const contractsDir = path.resolve(__dirname, '../contracts');
+process.chdir(contractsDir);
+
+// Verify each contract
+Object.entries(deployedContracts[network]).forEach(([contractName, contractInfo]: [string, any]) => {
+  const { address } = contractInfo;
+  
+  console.log(`Verifying ${contractName} on ${network}...`);
+  
+  try {
+    execSync(
+      `sncast verify --contract-address ${address} --contract-name ${contractName} --network ${network} --verifier walnut --confirm-verification` ,
+      { stdio: 'inherit' }
+    );
+    console.log(`Successfully verified ${contractName}`);
+  } catch (error) {
+    console.error(`Failed to verify ${contractName}:`, error);
+  }
+});
+
+console.log('Verification process completed.');
```

Comment:
Please wrap in a `main()` function, and wrap the `main()` function in the following wrapper. What this does is that it will ensure that this file only runs when u call it from the CLI, not through any imports.

```ts
function main() {
  // your logic here
}

if (typeof module !== "undefined" && require.main === module) {
  main()
}

```

Reply:
cc @gianalarcon @Nadai2010 @jrcarlos2000 we should enforce this standard too where applicable.

Reply:
Ready, the correction is applied, everything is working correctly.